polaris-react/src/components/AppProvider/AppProvider.tsx

```diff
@@ -24,6 +24,39 @@ import type {FeaturesConfig} from '../../utilities/features';
 import './AppProvider.scss';
 import './global.scss';
 
+function measureScrollbars() {
+  const maxScrollbarWidth = 20;
+  const overflowParentHeight = 30;
+  const overflowChildHeight = 40;
+  const parentEl = document.createElement('div');
+  parentEl.setAttribute(
+    'style',
+    `position: absolute; opacity: 0; transform: translate3d(-9999px, -9999px, 0); pointer-events: none; width:${overflowParentHeight}px; height:${overflowParentHeight}px;`,
+  );
+
+  const child = document.createElement('div');
+  child.setAttribute(
+    'style',
+    `width:100%; height: ${overflowChildHeight}; overflow:scroll`,
+  );
+  parentEl.appendChild(child);
+  document.body.appendChild(parentEl);
+
+  let scrollbarWidth =
+    overflowParentHeight - (parentEl.firstChild as HTMLDivElement).clientWidth;
```

Comment:
If you use [firstElementChild](https://developer.mozilla.org/en-US/docs/Web/API/Element/firstElementChild) instead of `firstChild` here, would we be able to get rid of the casting to `HTMLDivElement`?

Reply:
Good call! I got the linter to be happy by using `?? 0` after the null check.

Reply:
I think that should be fine and not something we'd ever hit. It would mean our test element didn't render and we we're unable to calculate the scrollbar, so we assume a scrollbar width of 0. However, if that component isn't rendering we got bigger problems than measuring the scrollbar.